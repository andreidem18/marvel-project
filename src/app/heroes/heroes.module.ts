import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { HeroesListComponent } from './components/heroes-list/heroes-list.component';
import { HeroeCardComponent } from './components/heroe-card/heroe-card.component';
import { SharedModule } from '../shared/shared.module';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { ComicsModule } from '../comics/comics.module';



@NgModule({
  declarations: [
    HeroesListComponent,
    HeroeCardComponent
  ],
  imports: [
    CommonModule,
    ReactiveFormsModule,

    ComicsModule,
    SharedModule,
  ],
  exports: [
    HeroesListComponent,
  ]
})
export class HeroesModule { }
