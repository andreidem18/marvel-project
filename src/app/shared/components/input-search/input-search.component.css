.input-search {
    width: 100%;
    font-size: 1.1rem;
    padding: 14px;
    padding-right: 50px;
    border-radius: 1px;
    border: none;
    display: block;
    position: relative;
    background: #fff url(../../../../assets/icons/search.png) no-repeat 98%;
}

