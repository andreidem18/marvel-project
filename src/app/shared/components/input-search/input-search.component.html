
<input 
    type="text" 
    placeholder="Search a character"
    class="input-search"
    [formControl]="inputSearch"
>

