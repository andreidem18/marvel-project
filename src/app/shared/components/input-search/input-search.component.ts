import { Component, OnInit, inject } from '@angular/core';
import { FormControl } from '@angular/forms';
import { debounceTime } from 'rxjs';
import { HeroesService } from '../../../heroes/services/heroes.service';

@Component({
  selector: 'shared-input-search',
  templateUrl: './input-search.component.html',
  styleUrls: ['./input-search.component.css']
})
export class InputSearchComponent implements OnInit {

  public inputSearch = new FormControl("");
  private heroesService = inject(HeroesService);

  ngOnInit(): void {
    this.inputSearch.valueChanges
      .pipe(debounceTime(500))
      .subscribe(value => {
        this.heroesService.getAllHeroes({ name: value || ""}).subscribe();
      });
  }

}
