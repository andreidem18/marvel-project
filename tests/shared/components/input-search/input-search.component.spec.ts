import { ComponentFixture, TestBed } from '@angular/core/testing';

import { InputSearchComponent } from '../../../../src/app/shared/components/input-search/input-search.component';
import { HttpClientTestingModule } from '@angular/common/http/testing';
import { ReactiveFormsModule } from '@angular/forms';
import { HeroesService } from 'src/app/heroes/services/heroes.service';

describe('InputSearchComponent', () => {
  let component: InputSearchComponent;
  let fixture: ComponentFixture<InputSearchComponent>;
  let compiled: HTMLElement;
  let service: HeroesService;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [HttpClientTestingModule, ReactiveFormsModule],
      declarations: [InputSearchComponent]
    });
    fixture = TestBed.createComponent(InputSearchComponent);
    service = TestBed.inject(HeroesService);
    component = fixture.componentInstance;
    compiled = fixture.nativeElement;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match with the snapshot', () => {
    expect(compiled.innerHTML).toMatchSnapshot();
  });

  it('should filter heroes by name when input', (done) => {
    jest.spyOn(service, 'getAllHeroes');
    const inputTag = compiled.querySelector('input');
    expect(inputTag).toBeTruthy();

    const heroName = 'spider-man'
    inputTag!.value = heroName;
    inputTag!.dispatchEvent(new Event('input'));

    setTimeout(() => {
      expect(service.getAllHeroes).toHaveBeenCalledWith({ name: heroName});
      done();
    }, 600);
  });
});
