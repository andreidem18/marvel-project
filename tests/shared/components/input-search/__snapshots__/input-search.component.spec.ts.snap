// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputSearchComponent should match with the snapshot 1`] = `"<input type="text" placeholder="Search a character" class="input-search ng-untouched ng-pristine ng-valid" ng-reflect-form="[object Object]">"`;
